CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+# Contributing to this repository
+
+# Audience
+This set of contributing guidelines is for both external contributors, as well as Verne Global employees. There are number of references to using JIRA issue number in this document. If you are an external contributor and do not have access to the Verne Global JIRA instance, please ignore these references.
+
+# Table of Contents
+1. [Commits and pull requests](#commits-release-notes-pull-requests)
+    1. [Fixing a bug](#fixing-a-bug)
+    2. [Commits](#commits)
+    3. [Pull requests](#pull-requests)
+        1. [Reviewing pull requests](#reviewing-pull-requests)
+        2. [Merging pull requests](#merging-pull-requests)
+
+## Commits and pull requests
+### Fixing a bug
+
+1. Clone the repo, and commit changes to a new branch, named using the convention <JIRA-ISSUE-NUMBER>-description-of-change. e.g. ```HPCDIR-9999-add-contributing```
+2. Make pull requests against the master branch in the repo.
+3. Reference the JIRA issue in both the commit message and the pull request message.
+
+### Commits
+
+Please avoid the following in a commit:
+
+* Mixing whitespace changes with functional code changes.
+* Mixing two unrelated functional changes in the same commit.
+* Sending large new features in a single commit.
+
+Expected git commit message structure (Based on [this article](https://chris.beams.io/posts/git-commit/)):
+
+* The first line should ideally be limited to 50 characters, but must be no more than 70 characters.
```

Comment:
In a different place, we had the ticket number in the first line of the commit message, like this:

    HPCDIR-1234:What this commit is about

This has two advantages over the currently proposed schemes:

  * It allows single-line commit messages, thus easy commits from the command line without the need to fire up an editor all the time.
  * After merging, you have a good overview over where the changes originated, without having to see each commit message in full.

Making committing easier helps making commits smaller, which is generally desirable.

Reply:
sounds good, and still works with the jira integrations. I'll update the doc